### PR TITLE
Nexus-18262: use api for version test including tests

### PR DESCRIPTION
**Pull request by jeremybryan**
Took the work initially completed by Kris Wright and added tests.  Kris modified the plugin here to use the Java api which does use proxy configuration if it's present. 

This has a corresponding set of changes in the platform-plugin as well.
https://github.com/sonatype/nexus-java-api/pull/109

JIRA: https://issues.sonatype.org/browse/NEXUS-18262

### CODE REVIEW

File: src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy [hunk 1]
```diff
@@ -35,51 +35,65 @@ class Nxrm3ConfigurationTest
...
-          return '<?xml version="1.0" encoding="UTF-8" standalone="yes"?><status><edition>PRO</edition><version>3.13' +
-              '.0-01</version></status>'
-        }
-        else {
-          throw new ConnectException()
-        }
-      }
+    when:
+      "checking $serverUrl for nxrm version"
+      client.getVersion() >> new com.sonatype.nexus.api.repository.v3.NxrmVersion(version, edition)
```
Comment (nmcafee) [inline src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy]: Question: Were you trying to make it clear which NxrmVersion you were dealing with here?

File: src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy [hunk 2]
```diff
@@ -140,14 +156,20 @@ class Nxrm3ConfigurationTest
   def 'defaults to anonymous access with no credentials'() {
     when:
       GroovySpy(Nxrm3Util.class, global: true)
+      client.getVersion() >> new com.sonatype.nexus.api.repository.v3.NxrmVersion(version, edition)
```
Comment (nmcafee) [inline src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy]: Question: Were you trying to make it clear which NxrmVersion you were dealing with here?

File: src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy [hunk 3]
```diff
@@ -35,51 +35,65 @@ class Nxrm3ConfigurationTest
...
-      'http://nxrm.valid.com'   | Kind.OK      | ''
-      'http://nxrm.valid.com/'  | Kind.OK      | ''
+      kind       | serverUrl        | credentialsId   | message
+      Kind.ERROR | 'http://foo.com' | 'credentialsId' | 'Nexus Repository Manager 3.x connection failed'
   }
 
   def 'it tests valid server credentials'() {
     when:
       client.getRepositories() >> repositories
+      client.getVersion() >> new com.sonatype.nexus.api.repository.v3.NxrmVersion(version, edition)
```
Comment (nmcafee) [inline src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy]: Question: Were you trying to make it clear which NxrmVersion you were dealing with here?

File: src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy (hunk 1 above)
Comment (jeremybryan) [reply src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy]: Yeah but I had intended to add the import and reduce this down. Thanks for the reminder! ;)

File: src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy (hunk 2 above)
Comment (jeremybryan) [reply src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy]: Yeah but I had intended to add the import and reduce this down. Thanks for the reminder! ;)

File: src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy (hunk 3 above)
Comment (jeremybryan) [reply src/test/java/org/sonatype/nexus/ci/config/Nxrm3ConfigurationTest.groovy]: Yeah but I had intended to add the import and reduce this down. Thanks for the reminder! ;)

File: pom.xml [hunk 4]
```diff
@@ -86,7 +86,7 @@
       <dependency>
         <groupId>com.sonatype.nexus</groupId>
         <artifactId>nexus-platform-api</artifactId>
-        <version>3.3.20181025-095610.1ce86ad</version>
+        <version>3.4-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (joedragons) [reply pom.xml]: Good eye @nmcafee , I saw this too manually testing.

**Comment by joedragons** [comment]
Just noticed CI is failing too?

**Comment by joedragons** [comment]
Waiting for this to be on RSO, then I think it'll pass.

**Comment by joedragons** [comment]
Tried to tweak something to make CI fire again post java-api availability but seems to have just stopped.  Asking.

**Pull request merged**